### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by karanshah-browserstack**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,69 @@
...
+
+
+**Note**: If you are facing any issues, refer [Getting Help section](#Getting-Help)
+
+## Integration with other Java frameworks
+
+For other Java frameworks samples, refer to following repositories :
+
+- [JUnit](https://github.com/browserstack/junit-appium-app-browserstack)
+- [Java](https://github.com/browserstack/java-appium-app-browserstack)
```
Comment (sriteja777) [inline README.md:63]: Do we want to add other sample repos testng, serenity?

File: android/src/test/resources/features/localtest/Local.feature [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Feature: BrowserStack Local Testing
+
+  Scenario: BStack Local Test - Can check tunnel working
+    When I start test on the Local Sample App
+    Then I should see "Up and running"
```
Comment (sriteja777) [inline android/src/test/resources/features/localtest/Local.feature]: new line

File: android/src/test/resources/allure.properties [hunk 3]
```diff
@@ -0,0 +1 @@
+allure.results.directory=target/allure-results
```
Comment (sriteja777) [inline android/src/test/resources/allure.properties]: Allure is like open source competitor to us right? So do we want to have allure integration in our repos?

File: ios/src/test/resources/features/localtest/Local.feature [hunk 4]
```diff
@@ -0,0 +1,5 @@
+Feature: BrowserStack Local Testing
+
+  Scenario: BStack Local Test - Can check tunnel working
+    When I start test on the Local Sample App
+    Then I should see "Up and running"
```
Comment (sriteja777) [inline ios/src/test/resources/features/localtest/Local.feature]: new line

**Comment by sriteja777** [comment]
Do we want to do it for Cucumber Junit5 and Cuucmber Junit4? as we have in Automate sample repos

File: .github/workflows/maven-workflow-run.yml [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+      - name: Run android mvn test
+        run: |
+          cd android
+          mvn compile
+          mvn test
+      - name: Run android mvn profile sample-local-test
+        run: |
+          cd android
+          mvn compile
+          mvn test -P sample-local-test
```
Comment (khrithik) [inline .github/workflows/maven-workflow-run.yml]: env variables?

File: android/browserstack.yml [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+# Read more about buildIdentifiers here -> https://www.browserstack.com/docs/automate/selenium/organize-tests
+buildIdentifier: '#${BUILD_NUMBER}' # Supports strings along with either/both ${expression}
+# Set `framework` of your test suite. Example, `testng`, `cucumber`, `cucumber-testng`
+# This property is needed to send test context to BrowserStack (test name, status)
+framework: cucumber-testng
+
+source: cucumber-java:appium-sample-main:v1.0
+
+# app: ./WikipediaSample.apk
+app: ./LocalSample.apk #For running local tests
```
Comment (khrithik) [inline android/browserstack.yml]: Do we need to keep local sample by default?

File: android/src/test/java/com/browserstack/stepdefs/StackDemoSteps.java [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+    private WebDriver driver;
+
+    @Before
+    public void setUp() throws MalformedURLException {
+        MutableCapabilities capabilities = new MutableCapabilities();
+        HashMap<String, String> bstackOptions = new HashMap<>();
+        bstackOptions.putIfAbsent("source", "cucumber-java:appium-sample-main:v1.0");
+        capabilities.setCapability("bstack:options", bstackOptions);
+        driver = new AndroidDriver(
+                new URL("https://hub.browserstack.com/wd/hub"), capabilities);
```
Comment (khrithik) [inline android/src/test/java/com/browserstack/stepdefs/StackDemoSteps.java:35]: Should we keep the URL to `localhost` or `127.0.0.1`

File: README.md (hunk 1 above)
Comment (karanshah-browserstack) [reply README.md:63]: Not addding for now.

File: android/src/test/java/com/browserstack/stepdefs/StackDemoSteps.java (hunk 7 above)
Comment (karanshah-browserstack) [reply android/src/test/java/com/browserstack/stepdefs/StackDemoSteps.java:35]: Kept it consistent with existing cucumber sample

File: android/browserstack.yml (hunk 6 above)
Comment (karanshah-browserstack) [reply android/browserstack.yml]: Fixed

**Review by khrithik — APPROVED** [review]

**Review by darshilgada24 — APPROVED** [review]

**Pull request merged**